tests/_test_utils/test_verify_waypoint_data.py

```diff
@@ -0,0 +1,98 @@
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+"""
+
+from mslib.utils.verify_waypoint_data import verify_waypoint_data
+
+
+def test_verify_xml_waypoint():
+    """
+    tests various XML waypoint data content
+    """
+    flight_track_with_waypoints = """<?xml version="1.0" encoding="utf-8"?>
+    <FlightTrack version="9.0.0">
+      <ListOfWaypoints>
+        <Waypoint flightlevel="233.0" lat="41.601070573320186" location="" lon="41.355120439498535">
+          <Comments></Comments>
+        </Waypoint>
+        <Waypoint flightlevel="374.0" lat="48.19354838709677" location="" lon="33.74841526975632">
+          <Comments></Comments>
+        </Waypoint>
+        <Waypoint flightlevel="372.0" lat="51.23623045499366" location="" lon="-34.20481757994082">
+          <Comments></Comments>
+        </Waypoint>
+        <Waypoint flightlevel="20.0" lat="37.037047471474835" location="" lon="-40.797295393717434">
+          <Comments></Comments>
+        </Waypoint>
+      </ListOfWaypoints>
+    </FlightTrack>"""
+
+    flight_track_empty = """<?xml version="1.0" encoding="utf-8"?>
+    <FlightTrack version="9.0.0">
+      <ListOfWaypoints>
+      </ListOfWaypoints>
+    </FlightTrack>"""
+
+    flight_track_incomplete = """<?xml version="1.0" encoding="utf-8"?>
+        <FlightTrack version="9.0.0">
+          <ListOfWaypoints>
+            <Waypoint flightlevel="233.0" lat="41.601070573320186">
+              <Comments></Comments>
+            </Waypoint>
+            <Waypoint flightlevel="374.0">
+              <Comments></Comments>
+            </Waypoint>
+            <Waypoint flightlevel="372.0" lat="51.23623045499366" location="" lon="-34.20481757994082">
+              <Comments></Comments>
+            </Waypoint>
+            <Waypoint flightlevel="20.0" lat="37.037047471474835" location="">
+              <Comments></Comments>
+            </Waypoint>
+          </ListOfWaypoints>
+        </FlightTrack>"""
+
+    flight_track_with_typo = """<?xml version="1.0" encoding="utf-8"?>
+        <FlightTrack version="9.0.0">
+          <ListOfWaypoints>
+            <Waypoint flightlevel="233.0"" lat="41.601070573320186" location="" lon="41.355120439498535">
+              <Comments></Comments>
+            </Waypoint>
+            <Waypoint flightlevel="374.0" lat="48.19354838709677" location="" lon="33.74841526975632">
+              <Comments></Comments>
+            </Waypoint>
+              <Comments></Comments>
+            </Waypoint>
+          </ListOfWaypoints>
+        </FlightTrack>"""
+
+    xml_content = (
+        (flight_track_with_waypoints, True),
+        (flight_track_empty, False),
+        (flight_track_incomplete, False),
+        (flight_track_with_typo, False),
+    )
```

Comment:
The pytest-idiomatic way for this would be to parametrize the test over these cases: https://docs.pytest.org/en/7.1.x/how-to/parametrize.html

Reply:
I know, but then the cases outside the function and I don't see that as better readable.

Reply:
Yes, they would be outside the function because each case would then actually be a separate test, as far as pytest is concerned. I find it more understandable when it is clear from the failing test itself which case fails, instead of having to look at the test function and figuring out in which loop iteration the assert fails.

Reply:
separating the tests by different names I can agree but for xml the fixture gets unreadable.

Reply:
I added one of the linebreak patterns too, which fails too.

Reply:
Fixture? What fixture?

Sorry, I don't see why something like this would be less readable:
```python
flight_track_with_waypoints = """<?xml version="1.0" encoding="utf-8"?>
<FlightTrack version="9.0.0">
  <ListOfWaypoints>
    <Waypoint flightlevel="233.0" lat="41.601070573320186" location="" lon="41.355120439498535">
      <Comments></Comments>
    </Waypoint>
    <Waypoint flightlevel="374.0" lat="48.19354838709677" location="" lon="33.74841526975632">
      <Comments></Comments>
    </Waypoint>
    <Waypoint flightlevel="372.0" lat="51.23623045499366" location="" lon="-34.20481757994082">
      <Comments></Comments>
    </Waypoint>
    <Waypoint flightlevel="20.0" lat="37.037047471474835" location="" lon="-40.797295393717434">
      <Comments></Comments>
    </Waypoint>
  </ListOfWaypoints>
</FlightTrack>"""

flight_track_empty = """<?xml version="1.0" encoding="utf-8"?>
<FlightTrack version="9.0.0">
  <ListOfWaypoints>
  </ListOfWaypoints>
</FlightTrack>"""

flight_track_incomplete = """<?xml version="1.0" encoding="utf-8"?>
<FlightTrack version="9.0.0">
  <ListOfWaypoints>
    <Waypoint flightlevel="233.0" lat="41.601070573320186">
      <Comments></Comments>
    </Waypoint>
    <Waypoint flightlevel="374.0">
      <Comments></Comments>
    </Waypoint>
    <Waypoint flightlevel="372.0" lat="51.23623045499366" location="" lon="-34.20481757994082">
      <Comments></Comments>
    </Waypoint>
    <Waypoint flightlevel="20.0" lat="37.037047471474835" location="">
      <Comments></Comments>
    </Waypoint>
  </ListOfWaypoints>
</FlightTrack>"""

flight_track_with_typo = """<?xml version="1.0" encoding="utf-8"?>
<FlightTrack version="9.0.0">
  <ListOfWaypoints>
    <Waypoint flightlevel="233.0"" lat="41.601070573320186" location="" lon="41.355120439498535">
      <Comments></Comments>
    </Waypoint>
    <Waypoint flightlevel="374.0" lat="48.19354838709677" location="" lon="33.74841526975632">
      <Comments></Comments>
    </Waypoint>
      <Comments></Comments>
    </Waypoint>
  </ListOfWaypoints>
</FlightTrack>"""

cases = [
    (flight_track_with_waypoints, True),
    (flight_track_empty, False),
    (flight_track_incomplete, False),
    (flight_track_with_typo, False),
]

@pytest.mark.parametrize("xml_content,verification_result", cases)
def test_verify_xml_waypoint(xml_content, verification_result):
    """Test xml verification."""
    assert verify_waypoint_data(xml_content) is verification_result
```

Reply:
I misunderstood you because you where pointing on the example.

Reply:
next update on the way

Reply:
Having it split into individual functions with descriptive names like you did looks good too though, pick what you prefer.

Reply:
compared to individual functions in a failure case that would look like

![image](https://github.com/user-attachments/assets/5ffb92d6-8201-4db0-83dd-612dc969d591)

How would I see now that flight_track_with_typo was the reason?

Reply:
A right, it shows the content of the variable as the test name by default. That can be changed with the `ids=` option to `pytest.mark.parametrize`.